### PR TITLE
pkg: Install all the dev-tools that are available

### PR DESCRIPTION
**Pull request by moyodiallo**
This is supposed to close https://github.com/ocaml/dune/issues/11001. 

But instead of having `dune tools install` I wonder if `dune build @pkg-tools` could be a better alternative, because this is equivalent building/invoking some aliases.

### CODE REVIEW

**Comment by maiste** [comment]
> I've added @maiste to weight in on this.

I would advocate for the opposite here. The aliases make sense when they are related to building targets in the context of Dune. Here, we are talking about development tools, so I'm afraid it would break the semantics as they are external tools. Also, in terms of accessibility, it would be easier for users to get access to the command than to the alias as with @gridbugs work, they have access to auto-completion.

My recommendation would be to stick to the command instead of the alias, but in the end, it is just my opinion :smile:

File: bin/lock_dev_tool.ml [hunk 1]
```diff
@@ -228,3 +228,10 @@ let lock_ocamlformat () =
 
 let lock_odoc () = lock_dev_tool Odoc None
 let lock_ocamllsp () = lock_dev_tool Ocamllsp None
+
+let lock_tools (tools : Dune_pkg.Dev_tool.t list) =
+  List.map tools ~f:(function
+    | Dune_pkg.Dev_tool.Odoc -> lock_odoc ()
```
Comment (Leonidas-from-XIV) [inline bin/lock_dev_tool.ml:234]: Can you add a type annotation so this prefix is not needed?

File: bin/tools/install_tools.ml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  let open Fiber.O in
+  let+ result =
+    let dev_tools_path =
+      List.map dev_tools ~f:(fun path -> Pkg_dev_tool.exe_path path |> Path.build)
+    in
+    Build_cmd.run_build_system ~common ~request:(fun _build_system ->
+      List.map dev_tools_path ~f:Action_builder.path |> Action_builder.all_unit)
+  in
+  match result with
+  | Error `Already_reported -> raise Dune_util.Report_error.Already_reported
```
Comment (Leonidas-from-XIV) [inline bin/tools/install_tools.ml:14]: I wonder, shouldn't this be `Code_error.raise`?

Comment (moyodiallo) [reply bin/tools/install_tools.ml:14]: If there is an error, it is already reported during the build and `Code_error.raise` is about raising with a message.

**Comment by gridbugs** [comment]
On the topic of installing dev tools with a command vs with an alias, I agree with @maiste that it breaks the semantics of aliases and is less accessible in terms of shell completion (at least until we can complete aliases which won't happen any time soon).

Looks like I accidentally duplicated this PR with https://github.com/ocaml/dune/pull/11146, which I closed because the overwhelming consensus among reviewers was that tools should be installed with an alias rather than a command. Thus I'm closing this too.

In the mean time individual tools can be installed by running them with `--help`, such as `dune tools exec ocamllsp -- --help`.

**Pull request closed without merging**